### PR TITLE
Escape html in table panel

### LINKED ISSUE #3673: Table: HTML Sanitize table content

### PR DESCRIPTION
**Pull request by bergquist**
By default all string in the table panel will be html escaped. 
It's possible to turn this off by setting colum style escapeHtml to false.

closes #3673

### CODE REVIEW

**Comment by torkelo** [comment]
think we should enforce html escaping , such a big security hole, at least it has to be disabled in the hosted version,

**Comment by bergquist** [comment]
Then ill remove the option

**Comment by bergquist** [comment]
@daniellee Would you please review this? always good to have a second opinion on security fixes :)

**Comment by daniellee** [comment]
Can you not just use [golang's EscapeString](https://golang.org/pkg/html/)? 

I compared your function to [golang's](https://golang.org/src/html/escape.go) and to  [.NET's htmlencode implementation](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/src/System/Net/WebUtility.cs#L52-L135) and it looks pretty similar except that it encodes some unicode values as well. Mostly symbols and non-latin letters (like öäå) from char 160 to 256 (see [here](https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references)) but more importantly a bunch of whitespace characters like nbsp. Do you think we need to do that too?

And finally should we escape the single quote character as well? See [owasp](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet#RULE_.231_-_HTML_Escape_Before_Inserting_Untrusted_Data_into_HTML_Element_Content)

**Comment by bergquist** [comment]
@daniellee we could copy golangs escapestring. But the escaping must be done in the frontend as long as we support direct requests to datasources. \ and ' could be added. That makes total sense. 

I don't think we need to encode unicode chars since we serve all requests as utf-8. But I might be misstaken :)

File: public/app/plugins/panel/table/renderer.ts [hunk 1]
```diff
@@ -141,4 +141,16 @@ export class TableRenderer {
...
   }
+
+  encodeHtml(unsafe) {
+    return unsafe.replace(/[&<>"']/g, function(m) {
+      return ({
+        '&': '&amp;',
+        '<': '&lt;',
+        '>': '&gt;',
+        '"': '&quot;',
+        '\'': '&#039;'
```
Comment (bergquist) [inline public/app/plugins/panel/table/renderer.ts]: escaped single quote chars

**Comment by daniellee** [comment]
Underscore/Lodash has an [escape method](https://github.com/lodash/lodash/blob/a0a649aaeabb0b99300ecb32f80f060f3f39815d/vendor/underscore/underscore.js#L1408-L1432) which is pretty identical to yours now :-)  Would you prefer to use that or the one you wrote?

**Comment by bergquist** [comment]
Ill change to using lodash.

**Pull request merged**